app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/metrics_capture.rb

```diff
@@ -0,0 +1,112 @@
+    "mem_usage_absolute_average" => {
+      :counter_key           => "mem_usage_absolute_average",
+      :instance              => "",
+      :capture_interval      => "20",
+      :precision             => 1,
+      :rollup                => "average",
+      :unit_key              => "percent",
+      :capture_interval_name => "realtime",
+    }.freeze,
+  }.freeze
+
+  def perf_collect_metrics(interval_name, start_time = nil, end_time = nil)
+    require 'rest-client'
+
+    raise _("No EMS defined") if ext_management_system.nil?
+
+    initialize_metrics_query(target.name)
+    counters_by_mor = {target.ems_ref => VIM_STYLE_COUNTERS}
+    counter_values_by_mor = {target.ems_ref => {}}
+
+    metrics_endpoint = ext_management_system.endpoints.find_by(:role => "metrics")
+    instance_id = metrics_endpoint.options["monitoring_instance_id"] unless metrics_endpoint.nil?
+    token = "Bearer " + IBMCloudSdkCore::IAMTokenManager.new(:apikey => ext_management_system.authentication_key("default")).access_token
+
+    response = RestClient::Request.execute(
+      :method  => :post,
+      :url     => "https://#{ext_management_system.provider_region}.monitoring.cloud.ibm.com/api/data",
+      :headers => {
+        'Content-Type'  => 'application/json',
+        'Authorization' => token,
+        'IBMInstanceID' => instance_id
+      },
+      :payload => File.open("metrics.json", 'rb')
+    )
+    data = JSON.parse(response.body)
+    dataset = consolidate_data(data["data"])
+
+    store_datapoints!(dataset[:timestamps], dataset[:cpu_usage_rate_average], "cpu_usage_rate_average", counter_values_by_mor[target.ems_ref])
+    store_datapoints!(dataset[:timestamps], dataset[:mem_usage_absolute_average], "mem_usage_absolute_average", counter_values_by_mor[target.ems_ref])
+    store_datapoints!(dataset[:timestamps], dataset[:net_usage_rate_average], "net_usage_rate_average", counter_values_by_mor[target.ems_ref])
+    store_datapoints!(dataset[:timestamps], dataset[:disk_usage_rate_average], "disk_usage_rate_average", counter_values_by_mor[target.ems_ref])
+
+    return counters_by_mor, counter_values_by_mor
+  rescue RestClient::ExceptionWithResponse => err
+    log_header = "[#{interval_name}] for: [#{target.class.name}], [#{target.id}], [#{target.name}]"
+    _log.error("#{log_header} Unhandled exception during perf data collection: [#{err}], class: [#{err.class}]")
+    _log.log_backtrace(err)
+    raise
+  end
+
+  def initialize_metrics_query(instance_name)
+    metrics_query = JSON.parse(File.read("metrics.json"))
+    metrics_query["filter"] = "ibm_resource_name = '#{instance_name}'"
+    File.write("metrics.json", JSON.pretty_generate(metrics_query))
+  end
+
+  def store_datapoints!(timestamps, datapoints, counter_key, counter_values_by_mor)
+    timestamps.zip(datapoints).each do |timestamp, datapoint|
+      counter_values_by_mor.store_path(Time.at(timestamp).utc, counter_key, datapoint)
+    end
+  end
+
+  def bps_to_kbps(datapoints)
```

Comment:
If you're modifying the argument (`.map!`) convention is to name this `bps_to_kbps!`